perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/GroupsManagerBlImpl.java

```diff
@@ -2313,6 +2323,34 @@ public RichGroup convertGroupToRichGroupWithAttributesByName(PerunSession sess,
 		return richGroups;
 	}
 
+	@Override
+	public List<RichGroup> convertGroupsToRichGroupsWithAttributes(PerunSession sess, Member member, Resource resource, List<Group> groups, List<String> attrNames) throws InternalErrorException, GroupResourceMismatchException, MemberResourceMismatchException {
+		List<RichGroup> richGroups = new ArrayList<>();
+		for(Group group: groups) {
+			List<Attribute> attributes = new ArrayList<>();
+
+			//Add all group and group-resource attributes (empty list means all possible attributes there)
+			List<String> supportedNames = attrNames.stream().filter(attrName ->
+				attrName.startsWith(AttributesManager.NS_GROUP_RESOURCE_ATTR) || attrName.startsWith(AttributesManager.NS_GROUP_ATTR)).collect(Collectors.toList()
+			);
+			attributes.addAll(getPerunBl().getAttributesManagerBl().getAttributes(sess, resource, group, supportedNames, true));
+
+			//Add all member-group attributes (empty list means all possible attributes there)
+			//WARNING - there is different behavior of method getAttributes for member and group by list of attr names, empty list means "no attributes to return"
+			supportedNames = attrNames.stream().filter(attrName ->
+				attrName.startsWith(AttributesManager.NS_MEMBER_GROUP_ATTR)).collect(Collectors.toList()
+			);
+			if(supportedNames.isEmpty()) {
```

Comment:
I'm not sure this is consistent with the rest of our API. Like this you always return all ``member-group`` attributes, even if caller specified eg. single ``group`` attribute in the outer API call.
So you should probably check emptiness only on outer ``attrNames`` param and not on filtered values. But maybe you have different expectations for this specific method.

Reply:
I changed this a bit. Now it should work fine.